src/tests/ftest/server/metadata.py

```diff
@@ -307,77 +262,279 @@ def test_metadata_server_restart(self):
+        self.delete_containers_in_array(container_array_less_ranks)
+        container_array, container_length = self.create_containers_until_error()
+        self.log.info("Containers created: %d. with one less rank"
+                      "metadata full)", container_length)
+
+        # Restore back the original system
+        dmg.system_start(ranks=pool_leader)
+
+    def test_metadata_pool_replica_leadership_change_space_free(self):
+        """
+        Test description:
+            Verify a pool leadership changes correctly when metadata is
+            not completely filled up.
+
+        Test steps:
+            Create a pool with storage replica(s) (using --nranks).
+            Query pool and save rank leader.
+            Fill metadata until DER_NOSPACE
+            Delete 10% of total containers
+            Change leadership (at the moment it should be done by stopping
+            the server that has the leader rank, for easy use, use 1 rank per
+            server).
+            Wait for rebuild to start -> complete.
+            Verify the pool is still serviceable:
+                Query pool
+                Attempt to add containers until DER_NOSPACE
+                Delete containers
+                Add containers again
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=small,hw
+        :avocado: tags=metadata_bug_with_debug
+        :avocado: tags=metadata_leader_change_space
+        """
+        test_result = True
+        dmg = self.get_dmg_command()
+        pool_leader = self.get_pool_leader()
+        self.log.info("Current leader: %s", pool_leader)
+
+        container_array, container_length = self.create_containers_until_error()
+
+        self.log.info("Deleting 50% of created containers")
+        percent = int(container_length * 0.50)
+        for cont in range(percent):
+            container_array[cont].destroy()
+            container_array.pop(cont)
+        self.log.info("Deleted 50% of containers")
+        self.log.info("Number of containers: %s", len(container_array))
+        dmg.system_stop(force=True, ranks=pool_leader)
+
+        from time import sleep
+        sleep(28800)  # Sleep for 8 hours
+
+        for attempt in range(5):
             try:
-                container = DaosContainer(self.context)
-                container.create(self.pool.pool.handle)
-                container_array.append(container)
-            except DaosApiError as exc:
-                self.log.info("(1.3)Expected DaosApiError info: %s", exc)
-                if der_no_space not in str(exc):
-                    self.fail(
-                        "##Expecting der_no_space RC: -1007, not seen")
-                else:
-                    self.log.info(traceback.format_exc())
-                    self.log.info("(1.4)No space error shown with additional "
-                                  "%d containers created, test passed.", cont)
+                if self.get_pool_rebuild_status == "idle":
```

Comment:
(pylint-comparison-with-callable) Comparing against a callable, did you omit the parenthesis?